tools/semgrep.yml

```diff
@@ -209,3 +209,27 @@ rules:
       Annotated types containing zerver.lib.typed_endpoint.ApiParamConfig should not be nested inside Optional. Use Annotated[Optional[...], zerver.lib.typed_endpoint.ApiParamConfig(...)] instead.
     languages: [python]
     severity: ERROR
+
+  - id: exists-instead-of-count
+    patterns:
+      - pattern-either:
+          - pattern: ... .count() == 0
+          - pattern: |
+              if not ... .count():
+                  ...
+    message: 'Use "not .exists()" instead; it is more efficient'
+    languages: [python]
+    severity: ERROR
+
+  - id: exists-instead-of-count-not-zero
+    patterns:
+      - pattern-either:
+          - pattern: ... .count() != 0
+          - pattern: ... .count() > 0
+          - pattern: ... .count() >= 1
+          - pattern: |
+              if ... .count():
+                  ...
+    message: 'Use ".exists()" instead; it is more efficient'
+    languages: [python]
+    severity: ERROR
```

Comment:
@andersk: Can you give these semgrep rules a gander?  I've not written many before, and while these don't seem to currently have any false-positives, it feels like they can possibly be tightened up.

I explicitly didn't include `.filter(...)` in the chain, since some of the callsites are just `ModelName.objects.count()`.  It also felt like it was difficult to handle all of `ModelName.objects.count()` / `variable.count()` and `class_variable._default_manager.count()`, but maybe there's a concise shortcut I'm not familiar with.

Reply:
Resolved on CZO as being an issue to fix later if it comes up.